wgsl/index.bs

```diff
@@ -4557,11 +4555,282 @@ TODO: *Stub*: Expression evaluation
+Similarly, `CF, E -> CF, V, B` means "run the analysis on expression E, starting with control-flow CF, apply the required changes to the graph, and name the resulting control-flow vertex CF, the resulting value vertex V, and the resulting set of behaviors B" (see next section for the analysis of expressions).
+
+We have a similar set of rules for expressions in left-value positions, that we denote by `CF, E => CF, L, B`. Instead of computing the vertex which corresponds to the uniformity of the value, it computes the vertex which corresponds to the uniformity of the variable we are adressing.
+
+<table class='data'>
+  <caption>Uniformity rules for statements</caption>
+  <thead>
+    <tr><th>Statement<th>New vertices<th>Recursive analyses<th>Behaviors<th>Resulting control-flow vertex<th>New edges
+  </thead>
+  <tr><td class="nowrap">*s1*; *s2*
+      <td>
+      <td class = "nowrap">*CF*, *s1* -> *CF1*, *B1*<br>
+          *CF1*, *s2* -> *CF2*, *B2*
+      <td>*B1* \ {Nothing} U *B2*
+      <td>*CF2*
+      <td>
+  <tr><td class="nowrap">if (*e*) then *s1* else *s2*
+      <td>*CFinside*, *CFend*
+      <td class="nowrap">*CF*, *e* -> *CF'*, *V*, *B*<br>
+          *CFinside*, *s1* -> *CF1*, *B1*<br>
+          *CFinside*, *s2* -> *CF2*, *B2*
+      <td>*B* U *B1* U *B2*
+      <td>*CFend*
+      <td class="nowrap">*CFinside* -> {*CF'*, *V*}<br>
+          *CFend* -> {*CF1*, *CF2*}
+  <tr><td class="nowrap">loop {*s1* continuing {*s2*}}
+      <td>*CF'*
+      <td class="nowrap">*CF'*, *s1* -> *CF1*, *B1*<br>
+          *CF1*, *s2* -> *CF2*, *B2*
+      <td>*B1* \ {Break, Continue} U *B2* U {Nothing}
+      <td>*CF'*
+      <td class="nowrap">*CF'* -> *CF2*
+  <tr><td class="nowrap">loop {*s1*}
+      <td>*CF'*
+      <td class="nowrap">*CF'*, *s1* -> *CF1*, *B1*
+      <td>*B1* \ {Break, Continue} U {Nothing}
+      <td>*CF'*
+      <td class="nowrap">*CF'* -> *CF1*
+  <tr><td class="nowrap">break;
+      <td>
+      <td>
+      <td>{Break}
+      <td>*CF*
+      <td>
+  <tr><td class="nowrap">continue;
+      <td>
+      <td>
+      <td>{Continue}
+      <td>*CF*
+      <td>
+  <tr><td class="nowrap">discard;
+      <td>
+      <td>
+      <td>{Discard}
+      <td>*CF*
+      <td>
+  <tr><td class="nowrap">return;
+      <td>
+      <td>
+      <td>{Return}
+      <td>*CF*
+      <td>*CF_return* -> *CF*
+  <tr><td class="nowrap">return *e*;
+      <td>
+      <td class="nowrap">*CF*, *e* -> *CF'*, *V*, *B*
+      <td>B \ {Nothing} U {Return}
+      <td>*CF'*
+      <td>*CF_return* -> *CF'*<br>
+          *Value_return* -> *V*
+  <tr><td class="nowrap">e1 = e2;
+      <td>
+      <td class="nowrap">*CF*, *e1* => *CF1*, *L1*, *B1*<br>
+          *CF1*, *e2* -> *CF2*, *V2*, *B2*
+      <td>B1 U B2
+      <td>*CF2*
+      <td>*L1* -> {*V2*, *CF*}
+</table>
```

Comment:
what about other statements? like a switch statement, or a function call?

Reply:
Yeah, comparing with the updated Behaviour analysis, looks like these are needed, but would be pretty mechanical (non-controversial) updates.
* empty compound statement: { }
* Non-empty compound statement { s1 }
* Rename "s1 ; s2" to "s1 s2"  
* Let-decl
* Var-decl
* Phony-assignment

Reply:
Most have been addressed.
Phony assignment is technically different from regular assignment, but should be straightforward.  No need to block on that.

Reply:
While I'm here, the switch statement should say that 'default' is treated like any other 'case' clause.  No need to block on that.

Reply:
Done, thanks.